server/resp/src/main/java/org/infinispan/server/resp/commands/set/SMOVE.java

```diff
@@ -0,0 +1,68 @@
+package org.infinispan.server.resp.commands.set;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.multimap.impl.EmbeddedSetCache;
+import org.infinispan.multimap.impl.SetBucket;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.AggregateCompletionStage;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * SMOVE implementation
+ *
+ * @since 15.0
+ */
+public class SMOVE extends RespCommand implements Resp3Command {
+   public SMOVE() {
+      super(4, 1, 2, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      var sourceK = arguments.get(0);
+      var destinationK = arguments.get(1);
+      EmbeddedSetCache<byte[], byte[]> esc = handler.getEmbeddedSetCache();
+      CompletableFuture<SetBucket<byte[]>> sourceStg, destinationStg;
+      sourceStg = esc.get(sourceK).toCompletableFuture();
+      destinationStg = esc.get(destinationK).toCompletableFuture();
+      final var element = arguments.get(2);
+      AggregateCompletionStage<Void> acs = CompletionStages.aggregateCompletionStage();
+      acs.dependsOn(sourceStg).dependsOn(destinationStg);
+      final var acsFroze = acs.freeze();
```

Comment:
I am really confused what is going on with this, it seems completely unneeded.

You can use `CompletionStage.thenCombine` to get a callback that is completed when both are done with their respective values.

Reply:
yeah, pretty a mess. let me see

Reply:
updated with a new version